### PR TITLE
SSCS-10388 Entity and Party Abstraction

### PR DESCRIPTION
**Pull request by TeaWithLucas**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SSCS-10388

### Change description ###

- [x] Abstract the Class Entity from the Appellant, Appointee, Representative and OtherParty classes.
- [x] Abstract the Class Party from the Appellant and OtherParty classes

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appellant.java [hunk 1]
```diff
@@ -1,44 +1,16 @@
 package uk.gov.hmcts.reform.sscs.ccd.domain;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.Builder;
-import lombok.Data;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appellant.java:6]: Please don't use experimental imports.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appellant.java [hunk 2]
```diff
@@ -1,44 +1,16 @@
...
 
 @JsonIgnoreProperties(ignoreUnknown = true)
 @Data
-@Builder(toBuilder = true)
-public class Appellant {
+@SuperBuilder
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appellant.java:10]: Needs to be removed and full class added back

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appointee.java [hunk 3]
```diff
@@ -1,32 +1,16 @@
 package uk.gov.hmcts.reform.sscs.ccd.domain;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.Builder;
-import lombok.Data;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.*;
+import lombok.experimental.SuperBuilder;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appointee.java:6]: Same here

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java [hunk 4]
```diff
@@ -0,0 +1,33 @@
+package uk.gov.hmcts.reform.sscs.ccd.domain;
+
+import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.RequiredArgsConstructor;
+import lombok.experimental.SuperBuilder;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java:8]: Same here

**Review by MohammedHMCTS — CHANGES_REQUESTED** [review]

**Review by wmarshid — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appointee.java [hunk 5]
```diff
@@ -1,32 +1,16 @@
 package uk.gov.hmcts.reform.sscs.ccd.domain;
 
-import com.fasterxml.jackson.annotation.JsonCreator;
 import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
-import com.fasterxml.jackson.annotation.JsonProperty;
-import lombok.Builder;
-import lombok.Data;
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.*;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Appointee.java:5]: This doesn't really matter but if you are not using everything in the library you should only import what you need also does would you need the line below if u are importing all?

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    private String organisation;
+
+    private YesNo confidentialityRequired;
+    private YesNo unacceptableCustomerBehaviour;
+    private YesNo vulnerableFlag;
+    private String vulnerabilityDetails;
+
+    private Subscription subscription;
+    private ReasonableAdjustmentDetails reasonableAdjustment;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java:32]: This is also a small one I think this should be renamed to resonableAdjustmentDetails just to make clear its the details as it sounds like it has YesNo resonableAdjustments

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/OtherParty.java [hunk 7]
```diff
@@ -1,49 +1,95 @@
...
 import static uk.gov.hmcts.reform.sscs.ccd.domain.YesNo.isYes;
 
-import com.fasterxml.jackson.annotation.JsonIgnore;
-import com.fasterxml.jackson.annotation.JsonInclude;
-import lombok.AllArgsConstructor;
-import lombok.Builder;
-import lombok.Data;
-import lombok.NoArgsConstructor;
+import com.fasterxml.jackson.annotation.*;
+import lombok.*;
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/OtherParty.java:7]: If we are not using everything in the library we should import what we need.

File: src/main/java/uk/gov/hmcts/reform/sscs/robotics/RoboticsJsonMapper.java [hunk 8]
```diff
@@ -76,11 +77,11 @@ private static JSONObject buildAppointeeDetails(Appointee appointee, boolean sam
         return buildContactDetails(json, appointee.getAddress(), appointee.getContact());
     }
 
-    private static JSONObject buildJointPartyDetails(JointPartyName jointPartyName, Address jointPartyAddress, boolean sameAddressAsAppellant,
+    private static JSONObject buildJointPartyDetails(Name name, Address jointPartyAddress, boolean sameAddressAsAppellant,
```
Comment (MohammedHMCTS) [inline src/main/java/uk/gov/hmcts/reform/sscs/robotics/RoboticsJsonMapper.java:80]: Doesn't matter much but jointPartyAddress should be just address if its following your pattern you have used above.

File: src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java (hunk 6 above)
Comment (TeaWithLucas) [reply src/main/java/uk/gov/hmcts/reform/sscs/ccd/domain/Entity.java:32]: that is its original name, id have to refactor more to change the name

File: src/test/java/uk/gov/hmcts/reform/sscs/ccd/validation/sscscasedata/JointPartyAddressAndAppellantAddressValidationTest.java [hunk 9]
```diff
@@ -4,10 +4,7 @@
 import javax.validation.ConstraintViolation;
 import org.junit.Assert;
 import org.junit.Test;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Address;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Appeal;
-import uk.gov.hmcts.reform.sscs.ccd.domain.Appellant;
-import uk.gov.hmcts.reform.sscs.ccd.domain.SscsCaseData;
+import uk.gov.hmcts.reform.sscs.ccd.domain.*;
```
[the remark this answers is not in the document]
Comment (TeaWithLucas) [reply src/test/java/uk/gov/hmcts/reform/sscs/ccd/validation/sscscasedata/JointPartyAddressAndAppellantAddressValidationTest.java:7]: 5 imports so ill leave the *

**Pull request merged**